test/testdata/infer/private_methods.rb

```diff
@@ -0,0 +1,66 @@
+# typed: true
+
+class Test
+  def using_symbol
+    1
+  end
+  private :using_symbol
+
+  private def using_symbol_returned_by_def
+    2
+  end
+
+  private def splat_call(*args)
+    puts args
+  end
+
+  private def splat_and_block_call(foo, bar, &blk)
+  end
+
+  private def block_call(&blk)
+  end
+
+  def calling_private
+    using_symbol
+    self.using_symbol
+    assigned_self = self
+    assigned_self.using_symbol # error: Non-private call to private method `Test#using_symbol`
+  end
+
+  private
```

Comment:
There are also private class methods defined like this:

```rb
class Foo
  class << self
    private def foo; end
  end
end

Foo.foo # error: Non-private call to private method `Foo#foo`
```

Reply:
Good call. Even though I would expect it to just work, we do need a bunch of tests for class methods as well